DiffPlex/DiffBuilder/InlineDiffBuilder.cs

```diff
@@ -113,6 +121,8 @@ private static void BuildDiffPieces(DiffResult diffResult, List<DiffPiece> piece
 
             for (; bPos < diffResult.PiecesNew.Length; bPos++)
                 pieces.Add(new DiffPiece(diffResult.PiecesNew[bPos], ChangeType.Unchanged, bPos + 1));
+
+            hasDifferences = pieces.Any(x => x.Type != ChangeType.Unchanged);
```

Comment:
Instead of needing the out parameter, why not just do this logic in the Diff method?